libkatana_python_native/src/Reductions.cpp

```diff
@@ -61,19 +61,20 @@ struct ReducibleFunctor {
         }),
         "Create a new instance with an initial value.");
     katana::RegisterNumbaClass(cls);
-    katana::DefWithNumba<const T&>(
-        cls, "update", &For::template type<T>::update,
+    katana::DefWithNumba<py::overload_cast<const T&>(
+        &For::template type<T>::update)>(
+        cls, "update",
         "Update this reducer with ``v`` performing the operation.");
-    katana::DefWithNumba<>(
-        cls, "reduce", &For::template type<T>::reduce,
+    katana::DefWithNumba<&For::template type<T>::reduce>(
```

Comment:
Ah so this is the change from the earlier conversation where you were passing in a reference ptr instead of just templating on the function itself?

Reply:
Yes. Though we are not templating on the function. We are templating on the function pointer. And it's suprisingly complex to get this working, much more template magic this way than with a normal argument.